### PR TITLE
Added the pytest fixtures

### PR DESCRIPTION
**Pull request by HeshamElsaman**
# Creating the pytest fixtures
## Linting
  - It 100% lints.

## Contents
  - I defined two functions that generate simulated grayscale images data for square and rectangular images.
  - The data are returned as a pandas data frame with the default indexing (it made since to me in realizing we'll be mainly interested in the periodicity of the positions of lattice points)
  - File Changed: test_preparation.py
  - Changes: the creation of two pytest fixtures

## Reviewer
  - I have no specific suggestion in mind.

### CODE REVIEW

**Comment by kylemasc917** [comment]
@HeshamElsaman Make sure to use the appropriate label and milestone in the side panel.

**Comment by kylemasc917** [comment]
@avgagliardo I've assigned you as reviewer for this request. Please do so as soon as possible.

**Comment by kylemasc917** [comment]
@avgagliardo Disregard Ive just seen you volunteer on another pull request.

**Comment by kylemasc917** [comment]
@iglesias-cardinale I've added you as reviewer for this request. Please do so as soon as possible.

**Comment by kylemasc917** [comment]
@iglesias-cardinale A reminder to review this as soon as possible.

**Comment by iglesias-cardinale** [comment]
Beginning my review now

File: test_preparation.py [hunk 1]
```diff
@@ -1,7 +1,33 @@
-""" unit test template """
+""" This is to test the functions created in the preparation module """
 
+import numpy as np
+import pandas as pd
+import pytest
 import preparation as prep
 
 def test_dummy():
```
Comment (iglesias-cardinale) [inline test_preparation.py:8]: We can probably remove the dummy test function since we have actual code in this file now.

**Review by iglesias-cardinale — APPROVED** [review]
I see that the fixtures are creating a square (256x256 pixel) and rectangular (512x256 pixel) random image and since you're randomly assigning a pixel value, the output should look like noise. I'mm approving the PR because it lints and tests and there are no problems with the code.

My main concern is how these will be used in actually testing the code that our peers will produce. Maybe you can add something to explain why these fixtures are useful?

**Comment by kylemasc917** [comment]
@HeshamElsaman I'll merge this now so others can use it for their functions but feel free to edit it and push that as another pull request. @iglesias-cardinale  Would you be willing to also review any further pull requests regarding these that H pushes through?

**Comment by HeshamElsaman** [comment]
@iglesias-cardinale Well, first, I guess for the test we don't care about what the data actually looks like as long as it's in a valid format and matches the expected/desired output. Second, I believe that how to use them will depend on the nature of the function. My suggestion is to store the output data in one local variable to be able to run the different operations on it. My only note is that my code generates Data frames that are spaced by the default indexing (which I noted in my comments and docstrings). So, technically, the frequency for this data would be inverse pixel. I believe we can keep using that as long as it's consistent throughout the assignment since also your code generates the arrays for the images which are also spaced in pixels not in meters. But that's an issue to discuss with Zach (@zbpetersbuf ) as well.
But if you're asking why they're useful in general, they're useful for that you'll be using this generated data to test your own functions.
My only issue will be with the frequency unit, but as long as you approve of it and will be able to use and modify the inverse pixel unit to your advantage, then all is set.

**Pull request merged**